src/game/console_cmd.c

```diff
@@ -0,0 +1,154 @@
+#include "game/console_cmd.h"
+
+#include "config.h"
+#include "game/console.h"
+#include "game/items.h"
+#include "game/lara.h"
+#include "game/random.h"
+#include "global/const.h"
+#include "global/types.h"
+#include "global/vars.h"
+
+#include <math.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#define ENDS_WITH_ZERO(num) (fabsf((num)-roundf((num))) < 0.0001f)
+
+static bool Console_Cmd_Pos(const char *const input)
+{
+    if (strcmp(input, "pos") == 0) {
+        if (!g_Objects[O_LARA].loaded) {
+            return true;
+        }
+        Console_Log(
+            "Room: %d\nPosition: %.3f, %.3f, %.3f\nRotation: %.3f,%.3f,%.3f ",
+            g_LaraItem->room_number + 1, g_LaraItem->pos.x / (float)WALL_L,
+            g_LaraItem->pos.y / (float)WALL_L,
+            g_LaraItem->pos.z / (float)WALL_L,
+            g_LaraItem->pos.x_rot * 360.0f / (float)PHD_ONE,
+            g_LaraItem->pos.y_rot * 360.0f / (float)PHD_ONE,
+            g_LaraItem->pos.z_rot * 360.0f / (float)PHD_ONE);
+        return true;
+    }
+
+    return false;
+}
+
+static bool Console_Cmd_Teleport(const char *const input)
+{
+    {
+        float x, y, z;
+        if (sscanf(input, "tp %f %f %f", &x, &y, &z) == 3) {
+            if (ENDS_WITH_ZERO(x)) {
+                x += 0.5f;
+            }
+            if (ENDS_WITH_ZERO(z)) {
+                z += 0.5f;
+            }
+
+            if (Item_Teleport(g_LaraItem, x * WALL_L, y * WALL_L, z * WALL_L)) {
+                Console_Log("Teleported to position: %.3f %.3f %.3f", x, y, z);
+                return true;
+            }
+
+            Console_Log(
+                "Failed to teleport to position: %.3f %.3f %.3f", x, y, z);
+            return true;
+        }
+    }
+
+    {
+        int16_t room_num = -1;
+        if (sscanf(input, "tp %hd", &room_num) == 1) {
+            if (!g_Objects[O_LARA].loaded || !g_LaraItem->hit_points) {
+                return true;
+            }
+            if (room_num < 1 || room_num > g_RoomCount) {
+                Console_Log(
+                    "Invalid room: %d. Valid rooms are 1-%d", room_num,
+                    g_RoomCount);
+                return true;
+            }
+            int16_t room_idx = room_num - 1;
```

Comment:
I'm wondering if 0-based room indexing will be better - it may just be personal preference, but I would tend to use this by checking the room number in trview then entering the command. Rooms aren't always next to each other sequentially so otherwise it could be a little confusing.

Reply:
Ideally I think we should be compatible with TRViewer and level builders mindsets and tooling. I kind of assumed it's 1-indexed. Do we know what TRViewer and TombEditor use?

Reply:
Yes, both are 0 based.

![image](https://github.com/LostArtefacts/TR1X/assets/33758420/46eff2b4-1221-444e-8b0c-5ea4d56619d3)
![image](https://github.com/LostArtefacts/TR1X/assets/33758420/f58c0f4b-6c6f-4eb1-9dde-c0e997226f2e)

Reply:
Thank you. Changed to 0 both for `/tp X` and `/pos`.